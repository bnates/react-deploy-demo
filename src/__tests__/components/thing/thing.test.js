import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Thing from '../../../components/thing/thing.js';

describe('<Thing />', () => {
  it('exists after my application has loaded', () => {
    let app = shallow(<Thing />);
    expect(app.find('p').exists()).toBeTruthy();
    expect(app.find('button').exists()).toBeTruthy();
  });

  it('changes the state of our component on click', () => {
    let app = mount(<Thing />);
    let button = app.find('button');
    button.simulate('click');

    expect(app.state('stuff')).toBe(false);
    expect(app.find('p').text()).toContain('false');
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Thing />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});