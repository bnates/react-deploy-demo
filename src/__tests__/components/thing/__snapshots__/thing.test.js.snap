// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thing /> renders correctly 1`] = `
<section
  className="thing"
>
  <p>
    Current Val: 
    true
  </p>
  <button
    onClick={[Function]}
  >
    Click to Change
  </button>
</section>
`;
